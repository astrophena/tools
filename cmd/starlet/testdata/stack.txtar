-- bot.star --
def handle(update):
    debug.stack()
    debug.go_stack()
    # Should crash.
    debug.stack(1)
-- update.json --
{
  "update_id": 10000,
  "message": {
    "date": 1441645532,
    "chat": {
      "last_name": "Test Lastname",
      "id": 123456789,
      "type": "private",
      "first_name": "Test Firstname",
      "username": "Testusername"
    },
    "message_id": 1365,
    "from": {
      "last_name": "Test Lastname",
      "id": 123456789,
      "first_name": "Test Firstname",
      "username": "Testusername"
    },
    "forward_from": {
      "last_name": "Forward Lastname",
      "id": 222222,
      "first_name": "Forward Firstname"
    },
    "forward_date": 1441645550,
    "text": "/start"
  }
}
