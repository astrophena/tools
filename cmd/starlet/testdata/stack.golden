[
  {
    "method": "sendMessage",
    "args": {
      "chat_id": 123456789,
      "entities": [
        {
          "length": 105,
          "offset": 25,
          "type": "pre"
        }
      ],
      "link_preview_options": {
        "is_disabled": true
      },
      "text": "‚ùå Something went wrong:\n\nTraceback (most recent call last):\n  bot.star:5:16: in handle\nError in debug.stack: unexpected arguments\n\n"
    }
  }
]